app/components/page-components/Chatbot/windows/ChatbotSongRequestOnboardingWindow.vue.ts

```diff
@@ -0,0 +1,103 @@
+import { Component, Prop } from 'vue-property-decorator';
+import ChatbotWindowsBase from 'components/page-components/Chatbot/windows/ChatbotWindowsBase.vue';
+import { $t } from 'services/i18n';
+import { cloneDeep } from 'lodash';
+import { IMediaShareSettings } from 'services/widget-settings/media-share';
+
+@Component({})
+export default class ChatbotSongRequestOnboardingWindow extends ChatbotWindowsBase {
+  // onboarding
+  step = 1;
+
+  // get from state media share service
+  settings: any = null;
+
+  async mounted() {
+    await this.chatbotApiService.fetchSongRequestPreferencesData();
+    this.settings = cloneDeep(this.chatbotApiService.state.songRequestPreferencesResponse.settings);
+  }
+
+  async onToggleNextHandler() {
+    if (this.step === 1) {
+      await this.chatbotApiService.updateSongRequestPreferencesData({settings: this.settings});
+      this.step++;
+    }
+    else this.chatbotCommonService.closeChildWindow();
+  }
+
+  onTogglePrevHandler() {
+    if (this.step > 1) this.step --;
+  }
+
+  get onboardingData() {
```

Comment:
Add a returning type for `onboardingData()` cuz it returns a complex object and it's very easy to make a mistake here